editable-list/main.js

```diff
@@ -1,132 +1,132 @@
 
-                this.itemList.appendChild(li);
-                this.itemList.children[childrenLength].appendChild(button);
+        li.textContent = textInput.value;
+        button.classList.add('editable-list-remove-item', 'icon');
+        button.innerHTML = '&ominus;';
 
-                this.handleRemoveItemListeners([...this.itemList.children]);
+        this.itemList.appendChild(li);
+        this.itemList.children[childrenLength].appendChild(button);
 
-                textInput.value = '';
-            }
-        }
+        this.handleRemoveItemListeners([...this.itemList.children]);
 
-        // fires after the element has been attached to the DOM
-        connectedCallback() {
-            const removeElementButtons = [...this.shadowRoot.querySelectorAll('.editable-list-remove-item')];
-            const addElementButton = this.shadowRoot.querySelector('.editable-list-add-item');
+        textInput.value = '';
+      }
+    }
 
-            this.itemList = this.shadowRoot.querySelector('.item-list');
+    // fires after the element has been attached to the DOM
+    connectedCallback() {
+      const removeElementButtons = [...this.shadowRoot.querySelectorAll('.editable-list-remove-item')];
+      const addElementButton = this.shadowRoot.querySelector('.editable-list-add-item');
 
-            this.handleRemoveItemListeners(removeElementButtons);
-            addElementButton.addEventListener('click', this.addListItem, false);
-        }
+      this.itemList = this.shadowRoot.querySelector('.item-list');
 
-        // gathering data from element attributes
-        get title() {
-            return this.getAttribute('title') || '';
-        }
+      this.handleRemoveItemListeners(removeElementButtons);
+      addElementButton.addEventListener('click', this.addListItem, false);
+    }
 
-        get items() {
-            const items = [];
+    // gathering data from element attributes
+    get title() {
+      return this.getAttribute('title') || '';
+    }
 
-            [...this.attributes].forEach(attr => {
-                if (attr.name.includes('list-item')) {
-                    items.push(attr.value);
-                }
-            });
+    get items() {
+      const items = [];
 
-            return items;
+      [...this.attributes].forEach(attr => {
+        if (attr.name.includes('list-item')) {
+          items.push(attr.value);
         }
+      });
 
-        get addItemText() {
-            return this.getAttribute('add-item-text') || '';
-        }
+      return items;
+    }
 
-        handleRemoveItemListeners(arrayOfElements) {
-            arrayOfElements.forEach(element => {
-                element.addEventListener('click', this.removeListItem, false)
-            });
-        }
+    get addItemText() {
+      return this.getAttribute('add-item-text') || '';
+    }
 
-        removeListItem(e) {
-            e.target.parentNode.remove();
-        }
+    handleRemoveItemListeners(arrayOfElements) {
```

Comment:
I was considering a PR to move the handlers from the elements themselves, to the surrounding `ul`. I think it would reduce the lines of code and complexity of this example pretty significantly.